### DISCUSSION TITLE
`weights_save_path` vs `ckpt_path`

### DISCUSSION THREAD

**Discussion by daniellepintz** [Lightning-AI/pytorch-lightning#10547] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Why do we need both `ckpt_path` as an arg to `fit`, and `weights_save_path` as a trainer constructor arg? Is there any way we can consolidate? We also have `default_root_dir`

cc @jjenniferdai

**Comment by jjenniferdai** [comment]
`ckpt_path` is for loading :) `weights_save_path` and `default_root_dir` are for saving

on a related note though, `default_root_dir` is used by both loggers and checkpoint, but `weights_save_path` I think is only used for checkpointing - can we deprecate `weights_save_path` from `Trainer` in favor of `ModelCheckpoint`'s `dirpath`?

**Reply by rohitgr7** [reply]
agreed!! This has been there for a while, way before when we split `filepath` into `filename` and `dirpath` in `ModelCheckpoint`. Plus, now we ask users to configure a callback instance anyways if they want to customize checkpoint callback.
